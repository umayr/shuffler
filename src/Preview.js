import React, { useState, useEffect } from 'react'

import { Box, Text, Heading, Button, Flex } from 'rebass'
import shortid from 'shortid'
import { useParams, useHistory } from 'react-router-dom'

import * as base64 from './base64'

function Preview() {
  const { state } = useParams()
  const history = useHistory()

  const [list, setList] = useState([])
  const [error, setError] = useState(null)

  useEffect(() => {
    try {
      setList(base64.from(state))
    } catch (err) {
      setError(err)
    }
  }, [state])

  const onClick = e => {
    e.preventDefault()

    const params = new URLSearchParams()
    list.forEach(item => params.append('item', item))

    history.push(`/?${params.toString()}`)
  }

  return (
    <Flex
      width={[1, 1, 1 / 2]}
      flexWrap="wrap"
      flexDirection="column"
      p={[4, 4, 5]}
    >
      <Box>
        {error === null && (
          <>
            <Box>
              {list.length > 0 &&
                list.map(item => (
                  <Heading
                    fontSize={[5, 6, 7]}
                    color="primary"
                    key={shortid.generate()}
                  >
                    {item}
                  </Heading>
                ))}
            </Box>
            <Box width={[1, 1, 1 / 2]} my={3}>
              <Button variant="primary" mr={2} onClick={onClick} width={1}>
                Shuffle
              </Button>
            </Box>
          </>
        )}
        {error !== null && (
          <>
            <Text fontSize={[3, 4, 5]} fontWeight="bold" color="primary">
              Invalid URL
            </Text>
            <Button variant="primary" mt={2}>
              Try Again
            </Button>
          </>
        )}
      </Box>
    </Flex>
  )
}

export default Preview
