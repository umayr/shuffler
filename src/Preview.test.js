import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Preview from './Preview'

import { MemoryRouter as Router, Route } from 'react-router-dom'

test('renders preview', () => {
  const { container } = render(<Preview />, { wrapper: Router })

  expect(container).toMatchSnapshot()
})

test('renders preview with error', () => {
  const { container } = render(<Preview />, { wrapper: Router })

  expect(container).toHaveTextContent('Invalid URL')
  expect(container).toHaveTextContent('Try Again')
})

test('shows preview with data', () => {
  const { container } = render(
    <Router
      initialEntries={[{ pathname: '/preview/WyJmb28iXQ==' }]}
      initialIndex={0}
    >
      <Route exact path="/preview/:state">
        <Preview />
      </Route>
    </Router>,
  )

  expect(container).toHaveTextContent(/foo/)
})
