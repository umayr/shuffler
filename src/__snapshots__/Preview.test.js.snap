// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders preview 1`] = `
<div>
  <div
    class="css-1fiyid0"
  >
    <div
      class="css-vurnku"
    >
      <div
        class="css-911jbo"
      >
        Invalid URL
      </div>
      <button
        class="css-wdk1eo"
      >
        Try Again
      </button>
    </div>
  </div>
</div>
`;
